typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -1,53 +1,599 @@
+      config,
+    });
+
+    // Deploy the InterchainGasPaymaster
+    const interchainGasPaymaster = await this.deployInterchainGasPaymaster({
+      storageGasOracle,
+      config,
+    });
+
+    return interchainGasPaymaster;
+  }
+
+  protected async deployInterchainGasPaymaster({
+    storageGasOracle,
+    config,
+  }: {
+    storageGasOracle: StorageGasOracle;
+    config: IgpConfig;
+  }): Promise<InterchainGasPaymaster> {
+    const deployerAddress = await this.multiProvider.getSignerAddress(
+      this.chainName,
+    );
+    const igp = await this.deployer.igpDeployer.deployProxiedContract(
+      this.chainName,
+      'interchainGasPaymaster',
+      'interchainGasPaymaster',
+      this.args.addresses.proxyAdmin,
+      [],
+      [deployerAddress, config.beneficiary],
+    );
+
+    const gasParamsToSet: InterchainGasPaymaster.GasParamStruct[] = [];
+    for (const [remote, gasOverhead] of Object.entries(config.overhead)) {
+      // TODO: add back support for non-EVM remotes.
+      // Previously would check core metadata for non EVMs and fallback to multiprovider for custom EVMs
+      const remoteDomain = this.multiProvider.tryGetDomainId(remote);
+      if (!remoteDomain) {
+        this.logger.warn(
+          `Skipping overhead ${this.chainName} -> ${remote}. Expected if the remote is a non-EVM chain.`,
+        );
+        continue;
+      }
+
+      this.logger.debug(
+        `Setting gas params for ${this.chainName} -> ${remote}: gasOverhead = ${gasOverhead} gasOracle = ${storageGasOracle.address}`,
+      );
+      gasParamsToSet.push({
+        remoteDomain,
+        config: {
+          gasOverhead,
+          gasOracle: storageGasOracle.address,
+        },
+      });
+    }
+
+    if (gasParamsToSet.length > 0) {
+      await this.multiProvider.handleTx(
+        this.chainName,
+        igp.setDestinationGasConfigs(
+          gasParamsToSet,
+          this.multiProvider.getTransactionOverrides(this.chainName),
+        ),
+      );
+    }
+
+    // Transfer ownership to the configured owner
+    await this.multiProvider.handleTx(
+      this.chainName,
+      igp['transferOwnership(address)'](
+        config.oracleKey,
+        this.multiProvider.getTransactionOverrides(this.chainName),
+      ),
+    );
+
+    return igp;
+  }
+
+  protected async deployStorageGasOracle({
+    config,
+  }: {
+    config: IgpConfig;
+  }): Promise<StorageGasOracle> {
+    const gasOracle = await this.deployer.igpDeployer.deployContract(
+      this.chainName,
+      'storageGasOracle',
+      [],
+    );
+
+    if (!config.oracleConfig) {
+      this.logger.debug('No oracle config provided, skipping...');
+      return gasOracle;
+    }
+
+    this.logger.info(`Configuring gas oracle from ${this.chainName}...`);
+    const configsToSet: Array<StorageGasOracle.RemoteGasDataConfigStruct> = [];
+
+    for (const [remote, desired] of Object.entries(config.oracleConfig)) {
+      // TODO: add back support for non-EVM remotes.
```

Comment:
is there an issue for this?

Reply:
my sense is we need to replace the multiprovider with one of 
- multi protocol provider
- registry 
- chain metadata manager or something here

Reply:
this comment was already there before

i think though that if the multiprovider is setup with chainmetadata that includes non-evm chains, then we're still fine - main issue is if we can't do the chainname<>domainId mapping not whether we have evm/non-evm

i think in infra before, we would automatically filter out non-evm chains, which is what led to the issue?